src/oop/object_independence/medium/q4/MainTest.java

```diff
@@ -0,0 +1,225 @@
+        assertTrue(testClass.hasModifier(secondFieldName, "private"),
+                "Your " + secondFieldName + " attribute does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of((Object) new String[]{"Pineapple Juice", "Coconut Milk"}),
+                Arguments.of((Object) new String[]{"Music1", "Music2"}),
+                Arguments.of((Object) new String[]{"Ginger Beer", "Lime", "Vodka"}),
+                Arguments.of((Object) new String[]{"Rum", "StrawBerry", "Syrup"}),
+                Arguments.of((Object) new String[]{"Pink Floyd", "Metallica"}),
+                Arguments.of((Object) new String[]{"Pop", "Rap"})
+        );
+    }
+
+    @Test
+    public void portableMusicPlayerClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {String[].class};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct modifier.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void portableMusicPlayerClassHasCorrectConstructor(String[] value) throws Throwable {
+        Object[][] arguments = {
+                {value, String[].class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        _assertArrayEquals(value, testClass.getFieldValue(classInstance, firstFieldName),
+                "Your " + className + " constructor does not correctly initialize the " + firstFieldName + " attribute.");
+        _assertEquals(0, testClass.getFieldValue(classInstance, secondFieldName),
+                "Your " + className + " constructor does not correctly initialize the " + secondFieldName + " attribute.");
+    }
+
+    @Test
+    public void playIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(firstMethod, null), String.format(incorrectMethodDefinition, firstMethod, className));
+        assertTrue(testClass.hasModifier(firstMethod, null, "public"), String.format(incorrectModifierMessage, firstMethod, className));
+        assertTrue(testClass.hasReturnType(firstMethod, null, Void.TYPE), String.format(incorrectReturnType, firstMethod, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void portableMusicPlayerClassHasCorrectPlayMethod(String[] value) throws Throwable {
```

Comment:
Unless I misunderstood, this only seems to test the play method when the current index is zero. You should also pass in some values for current to make sure it works with other indices.